build-llvm.py

```diff
@@ -576,13 +621,36 @@ def invoke_ninja(args, dirs, stage):
     if install_folder is not None:
         subprocess.run(['ninja', 'install'],
                        check=True,
-                       cwd=build_folder.as_posix(),
+                       cwd=build_folder,
                        stdout=subprocess.DEVNULL,
                        stderr=subprocess.DEVNULL)
 
         utils.create_gitignore(install_folder)
 
 
+def generate_pgo_profiles(dirs):
+    """
+    Build a set of kernels across a few architectures to generate PGO profiles
+    :param dirs: An instance of the Directories class with the paths to use
+    :return:
+    """
+
+    utils.print_header("Building PGO profiles")
+
+    # Run kernel/build.sh
+    subprocess.run([dirs.root_folder.joinpath("kernel", "build.sh")],
+                   check=True,
+                   cwd=dirs.build_folder.as_posix())
+
+    # Combine profiles
+    stage2_folder = dirs.build_folder.joinpath("stage2")
+    subprocess.run([
+        stage2_folder.joinpath("bin", "llvm-profdata"), "merge",
```

Comment:
Using the stage2 llvm-profdata may slow this step down since that is also built with instrumentation.  You can safely use the llvm-profdata from stage1.

Reply:
Thanks for the review, I’ll make that change tonight.